harness/determined/estimator/_estimator_trial.py

```diff
@@ -293,6 +294,48 @@ def control_loop(self) -> None:
             else:
                 raise AssertionError(f"Unknown wkld kind {wkld.kind}.")
 
+    def on_checkpoint_load(self, checkpoint_dir: str) -> None:
+        self.load_rng_state_from_checkpoint(checkpoint_dir)
+
+    def on_checkpoint_end(self, checkpoint_dir: str) -> None:
+        self.save_rng_state_with_checkpoint(checkpoint_dir)
+
+    def load_rng_state_from_checkpoint(self, checkpoint_dir: str) -> None:
+        rng_state = None
+        try:
+            with open(checkpoint_dir + "/rng_state.pkl", "rb") as f:
+                rng_state = pickle.load(f)
+        except IOError:
```

Comment:
can we add a comment here to say this is only for backwards compatibility? because I thought this logic should be removed sometime in the future

Reply:
Sure. Initially it was also to avoid problems on the first step of a new experiment, but now that it's in on_checkpoint_load() it is strictly for backwards compatibility.

I actually think this is much cleaner overall now that I'm using those 2 hooks.

Reply:
I also upgraded the log line to a warning again, since it's not supposed to happen on new experiments, and users of old checkpoints may want to be aware of this. I verified this code path is not active when running new experiments.

Reply:
thanks! can we add a prefix "backward compatibility: " in the comment as in other places?